compiler/native/environment.go

```diff
@@ -161,90 +160,62 @@ func (c *client) EnvironmentSecrets(s yaml.SecretSlice) (yaml.SecretSlice, error
-		"VELA_CHANNEL":        "TODO",
-		"VELA_DATABASE":       "TODO",
-		"VELA_DISTRIBUTION":   "TODO",
-		"VELA_HOST":           "TODO",
-		"VELA_NETRC_MACHINE":  "TODO",
-		"VELA_NETRC_PASSWORD": u.GetToken(),
-		"VELA_NETRC_USERNAME": "x-oauth-basic",
-		"VELA_QUEUE":          "TODO",
-		"VELA_RUNTIME":        "TODO",
-		"VELA_SOURCE":         "TODO",
-		"VELA_VERSION":        "TODO",
-		"VELA_WORKSPACE":      workspace,
-		"CI":                  "vela",
-
-		// repo specific environment variables
-		"REPOSITORY_BRANCH":    r.GetBranch(),
-		"REPOSITORY_CLONE":     r.GetClone(),
-		"REPOSITORY_FULL_NAME": r.GetFullName(),
-		"REPOSITORY_LINK":      r.GetLink(),
-		"REPOSITORY_NAME":      r.GetName(),
-		"REPOSITORY_ORG":       r.GetOrg(),
-		"REPOSITORY_PRIVATE":   unmarshal(r.GetPrivate()),
-		"REPOSITORY_TIMEOUT":   unmarshal(r.GetTimeout()),
-		"REPOSITORY_TRUSTED":   unmarshal(r.GetTrusted()),
-	}
-
-	// set tag environment variable if proper build event
-	if b.GetEvent() == constants.EventTag {
-		env["BUILD_TAG"] = strings.SplitN(b.GetRef(), "refs/tags/", 2)[1]
-	}
+	return combinedMap
+}
 
-	// set pull request number variable if proper build event
-	if b.GetEvent() == constants.EventPull {
-		env["BUILD_PULL_REQUEST_NUMBER"] = strings.SplitN(b.GetRef(), "/", 4)[2]
-	}
+// helper function that creates the standard set of environment variables for a pipeline.
+func environment(b *library.Build, m *types.Metadata, r *library.Repo, u *library.User) map[string]string {
+	workspace := "/vela"
 
-	// set deployment target if proper build event
-	if b.GetEvent() == constants.EventDeploy {
-		env["BUILD_TARGET"] = b.GetDeploy()
-	}
+	env := make(map[string]string)
+
+	// vela specific environment variables
+	env["VELA"] = library.ToString(true)
+	env["VELA_ADDR"] = "TODO"
+	env["VELA_CHANNEL"] = "TODO"
+	env["VELA_DATABASE"] = "TODO"
+	env["VELA_DISTRIBUTION"] = "TODO"
+	env["VELA_HOST"] = "TODO"
+	env["VELA_NETRC_MACHINE"] = "TODO"
+	env["VELA_NETRC_PASSWORD"] = u.GetToken()
+	env["VELA_NETRC_USERNAME"] = "x-oauth-basic"
+	env["VELA_QUEUE"] = "TODO"
+	env["VELA_RUNTIME"] = "TODO"
+	env["VELA_SOURCE"] = "TODO"
+	env["VELA_VERSION"] = "TODO"
+	env["CI"] = "vela"
 
 	// populate environment variables from metadata
 	if m != nil {
-		env["BUILD_CHANNEL"] = m.Queue.Channel
 		env["VELA_ADDR"] = m.Vela.WebAddress
 		env["VELA_CHANNEL"] = m.Queue.Channel
 		env["VELA_DATABASE"] = m.Database.Driver
 		env["VELA_HOST"] = m.Vela.Address
 		env["VELA_NETRC_MACHINE"] = m.Source.Host
 		env["VELA_QUEUE"] = m.Queue.Driver
 		env["VELA_SOURCE"] = m.Source.Driver
+		workspace = fmt.Sprintf("/vela/src/%s/%s/%s", m.Source.Host, r.GetOrg(), r.GetName())
+	}
+
+	env["VELA_WORKSPACE"] = workspace
+
+	// populate environment variables from repo library
+	mergeMap(env, r.Environment())
+	// populate environment variables from build library
+	mergeMap(env, b.Environment(workspace))
+	// populate environment variables from user library
+	mergeMap(env, u.Environment())
```

Comment:
Do we want these new user variables to be injected?

Reply:
@JordanSussman I could go either way 👍 

IMO, I don't think it hurts to have them there as the environment variables provided aren't considered sensitive.

That being said, maybe others will have a preference to not include the environment variables until they are requested?

Thoughts?

Reply:
I'd say just add them

Reply:
Opting to leave the new user variables in.